rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -291,6 +299,34 @@ fn base_connection_from_dbus(conn: &OwnedNestedHash) -> Option<BaseConnection> {
         base_connection.match_config = match_config_from_dbus(match_config)?;
     }
 
+    if let Some(ethernet_config) = conn.get(ETHERNET_KEY) {
+        if let Some(mac_address) = ethernet_config.get("assigned-mac-address") {
```

Comment:
Would be possible to move the processing of the assigned-mac-address to a separate function? Both branches look the same to me.